model/task/task.go

```diff
@@ -3495,9 +3495,53 @@ type StatusCount struct {
 	Count  int    `bson:"count"`
 }
 
-func GetTaskStatsByVersion(versionID string, opts GetTasksByVersionOptions) ([]*StatusCount, error) {
-	StatusCount := []*StatusCount{}
+type maxETAForQuery struct {
+	MaxETA time.Time `bson:"max_eta"`
+}
+
+type taskStatsForQueryResult struct {
+	Counts []StatusCount    `bson:"counts"`
+	ETA    []maxETAForQuery `bson:"eta"`
+}
```

Comment:
nit. these should live in the function since they are only used internally by it.